### PR TITLE
updated URLRequest WIP

### PR DESCRIPTION
**Pull request by homerblr**

### CODE REVIEW

File: Cinema Journal/Model/MainData.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+//
+//  Created by Homer on 10/21/20.
+//
+
+import Foundation
+
+struct MainData : Codable {
+    let page, totalResults, totalPages: Int
+      let results: [MovieDetails]
+
```
Comment (iKisliy) [inline Cinema Journal/Model/MainData.swift]: Indent (cmd + I)

File: Cinema Journal/Model/Networking.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//  Cinema Journal
+//
+//  Created by Homer on 10/21/20.
+//
+
+import Foundation
+
+struct Networking {
+    
+    static func fetchData() {
```
Comment (iKisliy) [inline Cinema Journal/Model/Networking.swift]: Now you need to make that method more useful. 
Create some convenient  data source for endpoints (paths). Personally, I prefer enum-based

**Review by iKisliy — CHANGES_REQUESTED** [review]

File: Cinema Journal/Model/MainData.swift (hunk 1 above)
Comment (homerblr) [reply Cinema Journal/Model/MainData.swift]: ctrl+I

File: Cinema Journal/Model/TVShowsResponse.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+//  TVShowsResponse.swift
+//  Cinema Journal
+//
+//  Created by Homer on 10/22/20.
+//
+
+import Foundation
+
+
+// MARK: - Welcome
```
Comment (iKisliy) [inline Cinema Journal/Model/TVShowsResponse.swift]: Keep it clean — It's not Welcome object

File: Cinema Journal/Model/Networking.swift [hunk 4]
```diff
@@ -9,33 +9,31 @@ import Foundation
...
-                print(jsonData.results[5].title)
+                let jsonData = try JSONDecoder().decode(T.self, from: data)
+                dump(jsonData)
             } catch {
                 print(error)
             }
-//            print(response)
             if let e = error {
                 print(e)
             }
```
Comment (iKisliy) [inline Cinema Journal/Model/Networking.swift]: OK, now you should return decoded object to caller of a function. Use completionHandler for that purpose
FYI: https://medium.com/@dhavalkansara51/completion-handler-in-swift-with-escaping-and-nonescaping-closures-1ea717dc93a4

**Review by iKisliy — CHANGES_REQUESTED** [review]

File: Cinema Journal/Model/Networking.swift [hunk 5]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 struct Networking {
     
-    static func fetchData<T>(_ request : APITarget, _ modelType : T.Type) where T: Codable {
+    static func fetchData<T>(_ request : APITarget, _ modelType : T.Type, completionHandler : @escaping ([String : Any]?, Error?) -> Void) where T: Codable {
```
Comment (iKisliy) [inline Cinema Journal/Model/Networking.swift]: Why downcasting to dictionary. You're using generics, you have `T.Type`, when you decode, it'll be `T`. So completion may be `@escaping (T?, Error?) -> Void`

File: Cinema Journal/Model/Networking.swift [hunk 6]
```diff
@@ -26,9 +26,9 @@ struct Networking {
             guard let data = data, let response = response else { return }
             do {
                 let jsonData = try JSONDecoder().decode(T.self, from: data)
-                dump(jsonData)
+                completionHandler(jsonData as? [String : Any], nil)
```
Comment (iKisliy) [inline Cinema Journal/Model/Networking.swift]: `completionHandler(jsonData, nil)`

**Review by iKisliy — CHANGES_REQUESTED** [review]
Use generics, later I'll show you Result type

File: Cinema Journal/Model/Networking.swift (hunk 6 above)
Comment (homerblr) [reply Cinema Journal/Model/Networking.swift:29]: Error: "_Cannot convert value of type 'T' to expected argument type '[String : Any]?_'"

Comment (iKisliy) [reply Cinema Journal/Model/Networking.swift]: Firstly change completionHandler at function signature like I pointed earlier

Comment (homerblr) [reply Cinema Journal/Model/Networking.swift]: Got it

File: Cinema Journal/Model/Networking.swift [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        urlComponents.path = request.path
+        urlComponents.queryItems = request.queryItems
+        
+        guard let stringURL = urlComponents.url?.absoluteString,
+              let url = URL(string: stringURL) else { return }
+        
+        //Networking
+        let session = URLSession.shared
+        session.dataTask(with: url) { (data, response, error) in
+            guard let data = data, let response = response else { return }
```
Comment (iKisliy) [inline Cinema Journal/Model/Networking.swift:26]: Completion will not be called, before return call completion with error

File: Cinema Journal/Model/Networking.swift [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            do {
+                let jsonData = try JSONDecoder().decode(T.self, from: data)
+                completionHandler(jsonData, nil)
+            } catch {
+                completionHandler(nil, error)
+            }
+            if let e = error {
+                print(e)
```
Comment (iKisliy) [inline Cinema Journal/Model/Networking.swift]: Same here. No completion, sender will be waiting forever

**Review by iKisliy — APPROVED** [review]
It's ok now. Tune flow with completion, and I beg you to try Result type instead of returning two optionals (T?, Error?) — https://www.hackingwithswift.com/articles/161/how-to-use-result-in-swift

**Pull request merged**